service/history/replication/executable_history_task.go

```diff
@@ -297,3 +302,145 @@ func (e *ExecutableHistoryTask) getDeserializedEvents() (_ []*historypb.HistoryE
 	}
 	return events, newRunEvents, err
 }
+
+func (e *ExecutableHistoryTask) BatchWith(incomingTask BatchableTask) (_ TrackableExecutableTask, retErr error) {
+	if !e.wantToBatch {
```

Comment:
so wanttobanch is one way，it doesn't support false to true I guess ？

Reply:
Yes you are right.